driver-sync/src/test/functional/com/mongodb/client/AbstractUnifiedTest.java

```diff
@@ -463,6 +471,42 @@ public Void execute() {
         }
     }
 
+    private void assertCollectionExists(final BsonDocument operation, final boolean shouldExist) {
+        BsonDocument arguments = operation.getDocument("arguments", new BsonDocument());
+        String databaseName = arguments.getString("database").getValue();
+        String collection = arguments.getString("collection").getValue();
+        assertEquals(shouldExist, collectionExists(databaseName, collection));
+    }
+
+    private boolean collectionExists(final String databaseName, final String collectionName) {
+        try (MongoClient client = MongoClients.create(MongoClientSettings.builder()
+                .applyConnectionString(connectionString).build())) {
+            return client.getDatabase(databaseName).listCollectionNames().into(new ArrayList<String>()).contains(collectionName);
+        }
+    }
+
+    private void assertIndexExists(final BsonDocument operation, final boolean shouldExist) {
+        BsonDocument arguments = operation.getDocument("arguments", new BsonDocument());
+        String db = arguments.getString("database").getValue();
+        String collection = arguments.getString("collection").getValue();
+        String index = arguments.getString("index").getValue();
+        assertEquals(shouldExist, indexExists(db, collection, index));
+    }
+
+    private boolean indexExists(final String databaseName, final String collectionName, final String indexName) {
+        try (MongoClient client = MongoClients.create(MongoClientSettings.builder()
```

Comment:
Can we use com.mongodb.client.Fixture#getMongoClient for this instead of creating a new instance each time?